File: MovieApp/MovieApp/Models/Movie.cs
using System;
using System.Collections.Generic;

namespace MovieApp.Models;

public partial class Movie
{
    public int MovieId { get; set; }

    public string Title { get; set; } = null!;

    public string Overview { get; set; } = null!;

    public string Genre { get; set; } = null!;

    public string Language { get; set; } = null!;

    public int Duration { get; set; }

    public decimal? Rating { get; set; }

    public string? PosterPath { get; set; }
}

File: MovieApp/MovieApp/ClientApp/src/app/models/movie.ts
export interface Movie {
  movieId: number;
  title: string;
  overview: string;
  genre: string;
  language: string;
  duration: number;
  rating: number;
  posterPath: string;
}

export type MovieType = {
  movieList: Movie[];
};

export type SimilarMovieType = {
  similarMovies: Movie[];
};

export type WatchlistType = {
  watchlist: Movie[];
};

export type ToggleWatchlistType = {
  toggleWatchlist: Movie[];
};
